src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -360,6 +410,25 @@ private async ValueTask<FlushResult> ProcessDataWrites()
             return flushResult;
         }
 
+        private Memory<byte> GetFakeMemory(int sizeHint)
```

Comment:
Maybe we should consider a shared output producer base class as there are now quite a few methods that are "shared".

Reply:
There's definitely some potential for better reuse. It could be a nice refactoring/cleanup down the road. I don't think it's critical for this change as this only duplicates a little more than we already were, and the logic that was copied is quite simple.